src/tqec/templates/display.py

```diff
@@ -29,6 +30,74 @@ def display_template(
         print()
 
 
+def display_templates_ascii(
+    templates: ComposedTemplate,
+    plaquette_indices: ty.Sequence[int] | None = None,
+) -> None:
+    """Display the templates as an ASCII art (multi-line string).
+
+    Args:
+        templates: the ComposedTemplate instance to display.
+        *plaquette_indices: the plaquette indices that are forwarded to the call
+            to `template.instantiate` to get the actual template representation.
+    """
+    if plaquette_indices is None:
+        plaquette_indices = tuple(range(1, templates.expected_plaquettes_number + 1))
+    # Numpy array including all templates.
+    arr = templates.instantiate(plaquette_indices)
+    # Upper-left and bottom-right corners of every template, expressed as (row, column) of the array returned.
+    ul_pos = {}
+    br_pos = {}
+    ul_positions = templates._compute_ul_absolute_position()
+    bbul, _ = templates._get_bounding_box_from_ul_positions(ul_positions)
+    for tid, tul in ul_positions.items():
+        # We follow the same procedure as in Template.instantiate():
+        # Subtracting bbul (upper-left bounding box position) from each coordinate to stick
+        # the represented code to the axes and avoid having negative indices.
+        x = tul.x - bbul.x
+        y = tul.y - bbul.y
+        # Recall that numpy indexing is (y, x) in our coordinate system convention.
+        tshapey, tshapex = templates._templates[tid].shape.to_numpy_shape()
+        ul_pos[tid] = (y, x)
+        br_pos[tid] = (y + tshapey, x + tshapex)
+    # Format of the ASCII art.   
+    h_space = 5
+    empty_line = ' '*numpy.shape(arr)[1]*h_space
```

Comment:
I would create a named variable for `numpy.shape(arr)[1]`, just to make that more explicit. Also, the convention used in the code base is to enclose strings in `"` instead of `'`:
```suggestion
    x_size = numpy.shape(arr)[1]
    empty_line = " "*x_size*h_space
```